sdks/python/apache_beam/transforms/util.py

```diff
@@ -825,7 +827,60 @@ def expand(self, pcoll):
               key_value[1]))
       return (
           sharded_pcoll
-          | GroupIntoBatches(self.batch_size, self.max_buffering_duration_secs))
+          | GroupIntoBatches(
+              self.params.batch_size, self.params.max_buffering_duration_secs))
+
+    def to_runner_api_parameter(
+        self,
+        unused_context  # type: PipelineContext
+    ):  # type: (...) -> Tuple[str, beam_runner_api_pb2.GroupIntoBatchesPayload]
+      return (
+          common_urns.composites.GROUP_INTO_BATCHES_WITH_SHARDED_KEY.urn,
+          self.params.get_payload())
```

Comment:
We can get ride of `GroupIntoBatchesParams` by creating `GroupIntoBatchesPayload ` directly here and accessing the payload directly below.

Reply:
The main reason I extracted the params to a class was to reuse the code as much as possible since the params of GroupIntoBatches and WithShardedKey are exactly the same. The class also does parameter validation other than conversion from/to runner api, and perhaps it would make it easier to add more params in the future. But let me know if you have opinions.

Reply:
I see. Thanks for the explanation.